builders/gradle.go

```diff
@@ -58,14 +58,20 @@ func (builder *GradleBuilder) Analyze(m module.Module, allowUnresolved bool) ([]
 	moduleConfigurationKey := strings.Split(m.Name, ":")
 	taskConfiguration := "compile"
 	taskName := moduleConfigurationKey[0]
-	if len(moduleConfigurationKey) == 2 {
+	if taskName != "" {
+		// this is not the root gradle project
+		taskName = taskName + ":"
```

Comment:
1. Avoid mutation, it makes your intent much less clear.
2. I'd locate this closer to where the variable is used, so it's clear that this is formatting for usage as an argument to `gradle`.